### ISSUE TITLE
Implement exchange Kraken

### ISSUE THREAD

**Issue by shredEngineer** [provinzio/CoinTaxman#8]
For Kraken, there are two [different](https://support.kraken.com/hc/en-us/articles/115000302707-Differences-between-ledger-and-trades-history) CSV export options for account statements: _Trades_ and _Ledgers_ history.

We need to use the _Ledgers_ history, as its format is vastly more useful for our purposes.

The transaction fees are listed in-line (and not in separate follow-up lines), so I think for each line the read-in function will have to generate _two_ `Operation`s (in the case of a non-zero fee).

Historical prices (OHLC data) and trade data can easily be fetched using the public API: https://support.kraken.com/hc/en-us/articles/218198197-How-to-retrieve-historical-time-and-sales-trading-history-using-the-REST-API-Trades-endpoint-

I'm currently working on this feature.

**Comment by provinzio** [comment]
We should state this somewhere in the Wiki https://github.com/provinzio/CoinTaxman/wiki.

I am not sure if you are allowed to edit the Wiki. If not, please write an example wiki entry for Kraken and how to obtain the correct history. (see https://github.com/provinzio/CoinTaxman/wiki/Exchange:-Binance)

**Comment by shredEngineer** [comment]
> We should state this somewhere in the Wiki https://github.com/provinzio/CoinTaxman/wiki.
> 
> I am not sure if you are allowed to edit the Wiki. If not, please write an example wiki entry for Kraken and how to obtain the correct history. (see https://github.com/provinzio/CoinTaxman/wiki/Exchange:-Binance)

Great! :+1:   I didn't know the Wiki existed, to be honest.
In the README, you should definitely link every exchange name in the list of Currently supported exchanges to the corresponding Wiki page.

Generally, it doesn't seem like I'm allowed to edit the Wiki. I'll make sure to provide you with the Wiki entry later.

> EDIT: And it would be awesome if the software gives a specific warning if you include the wrong file instead of saying "can not recognize file".

Will do, good idea!

**Comment by shredEngineer** [comment]
Wiki entry for _Exchange: Kraken_: [Exchange:-Kraken.md.txt](https://github.com/provinzio/CoinTaxman/files/6012689/Exchange.-Kraken.md.txt)

**Comment by provinzio** [comment]
I added your wiki entry and referenced them in the README file. We can remove the "Work in Progress" in the README with your PR.

**Comment by shredEngineer** [comment]
Here are some further notes about my experience with the Kraken API and the exported account statements so far:

- The OHLC data offered by the OHLC API endpoint does indeed **not** reach back infinitely far, so we have to read exchange rates directly from the trades history API endpoint. This is quite cumbersome, but it seems to work OK.
- In contrast to what is done for the binance case where prices are averaged (weighted by volume) over a 1-minute interval, in the Kraken case we only select the single data point closest to the desired "target" timestamp, but not newer than the target timestamp.
- For this we fetch one chunk of the trade history, starting 10 minutes before the target timestamp.
- We then walk through the history until the desired closest timestamp match is found. Otherwise, we start another 10 minutes earlier and try again, etc. … (Exiting with a warning and zero price after hitting the arbitrarily chosen offset limit of 120 minutes.)

----------

- On the other hand, the _Trades_ history **does** contain the exact ready-to-use exchanges rates, but **only** for the base currency and not for the quote currency; so for crypto-to-crypto trading, there's still a missing variable -- as there's not account balance field in the Trades history, we also cannot deduce the missing variable from that file. Also, the fees stated therein are only _estimated_; only the _Ledgers_ history contains the correct fees.
- **But** trades done via the "Buy Crypto" button don't seem to show up in the _Trades_ history, only in the _Ledgers_ history.

----------

- To summarize: In the future, we may decide to process _both_ files, in order to get correct fees **and** more exact exchange rates (for all transactions except "Buy Crypto" button trades, and still with one unknown exchange rate for crypto-to-crypto trades). However, for now we'll stick only with the Ledgers export, for reasons of simplicity. (Otherwise we'd have to ensure matching pairs of files are present in the `account_statements/` folder and synchronize / merge / process those files.)
